### PR TITLE
Add Baichuan2 in Basic Application Development

### PR DESCRIPTION
**Pull request by ch1y0q**
Add Baichuan2 in Basic Application Development. Change README.md and naming of `3_BasicApp.ipynb` accordingly

### CODE REVIEW

File: ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3.2 Load a pretrained Model\n",
+    "\n",
+    "Before using a LLM, you need to first load one. Here we take a relatively small LLM, i.e. [Baichuan2-7b-chat](https://huggingface.co/baichuan-inc/Baichuan2-7B-Chat) as an example.\n",
```
Comment (plusbang) [inline ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb:38]: 'take a relatively small LLM, i.e. [Baichuan2-7b-chat]' -> 'take [Baichuan2-7b-chat]'

File: ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> **Note**\n",
+    ">\n",
+    "> * If you want to use precisions other than INT4(e.g. NF4/INT5/INT8,etc.), or know more details about the arguments, please refer to [API document](https://ipex-llm.readthedocs.io/en/latest/doc/PythonAPI/LLM/transformers.html) for more information. \n",
```
Comment (plusbang) [inline ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb:79]: 'NF4/INT5/INT8' -> 'FP8/INT8'

**Review by plusbang — COMMENTED** [review]
other LGTM

File: ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install --pre --upgrade ipex-llm[all]"
+   ]
```
Comment (Oscilloscope98) [inline ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb:29]: @plusbang  Should we also add a notebox for how to install on Linux OS?

**Review by Oscilloscope98 — APPROVED** [review]
Others LGTM

File: ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb (hunk 3 above)
Comment (Oscilloscope98) [reply ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb:29]: See here for more info: https://ipex-llm.readthedocs.io/en/latest/doc/LLM/Overview/install_cpu.html#quick-installation

Comment (plusbang) [reply ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb:29]: > @plusbang Should we also add a notebox for how to install on Linux OS?

yeah, I think we could add this as a note.

File: ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb [hunk 4]
```diff
@@ -28,6 +28,15 @@
     "!pip install --pre --upgrade ipex-llm[all]"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> **Note**\n",
+    ">\n",
+    "> Please refer to https://ipex-llm.readthedocs.io/en/latest/doc/LLM/Overview/install_cpu.html#quick-installation for more details on how to set up the environment."
```
Comment (plusbang) [inline ch_3_AppDev_Basic/3_Baichuan2_BasicApp.ipynb:37]: It maybe better to remind users that: 

"On Linux OS,  we recommend to use `pip install --pre --upgrade ipex-llm[all] --extra-index-url https://download.pytorch.org/whl/cpu` to install. Please refer to https://ipex-llm.readthedocs.io/en/latest/doc/LLM/Overview/install_cpu.html#quick-installation for more details."

**Pull request merged**